File: UI/src/app/models/request/requestAuth.ts
// Auth request payloads — mirrors UI/src-tauri/src/models/request/request_auth.rs.

export interface requestLogin {
  email: string;
  password: string;
}

// Shared by every "email a one-time code" call — sendOtp, forgotPasswordSendOtp.
export interface requestEmail {
  email: string;
}

export interface requestVerifyOtp {
  email: string;
  otpCode: string;
}

export interface requestChangePassword {
  oldPassword: string;
  newPassword: string;
}

export interface requestRequestAccess {
  firstName: string;
  lastName: string;
  email: string;
  password: string;
  phoneNumber?: string;
  companyName?: string;
  otpCode: string;
}

File: UI/src-tauri/src/models/request/request_auth.rs
// Auth request payloads.
use serde::Deserialize;

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct RequestLogin {
    pub email: String,
    pub password: String,
}

// Shared by every "email a one-time code" command — send_otp,
// forgot_password_send_otp — since the shape is identical.
#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct RequestEmail {
    pub email: String,
}

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct RequestVerifyOtp {
    pub email: String,
    pub otp_code: String,
}

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct RequestChangePassword {
    pub old_password: String,
    pub new_password: String,
}

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct RequestRequestAccess {
    pub first_name: String,
    pub last_name: String,
    pub email: String,
    pub password: String,
    pub phone_number: Option<String>,
    pub company_name: Option<String>,
    pub otp_code: String,
}
